src/field/pokemon.ts

```diff
@@ -4044,6 +4064,47 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
     return true;
   }
 
+  protected setStatus(
+    effect: StatusEffect,
+    { toxicTurnCount = 0, sleepTurnsRemaining = 0 }: { toxicTurnCount?: number; sleepTurnsRemaining?: number },
+  ): void {
+    this.status = {
+      effect,
+      toxicTurnCount,
+      sleepTurnsRemaining,
+    };
+  }
+
+  public advanceStatusCounter(): void {
+    if (!this.status) {
```

Comment:
When is status not set, it's a required param, right?
I'd prefer to use a more descriptive condition here:
```suggestion
    if (this.status === StatusEffect.NONE) {
```

Reply:
Currently `Pokemon#status` is set to `null` if the pokemon doesn't have a status effect, so that wouldn't work.